SignalMessaging/utils/ConversationStyle.swift

```diff
@@ -136,76 +136,131 @@ public class ConversationStyle: NSObject {
             switch outgoingMessage.messageState {
             case .failed:
-                return self.bubbleColorOutgoingUnsent
+                return bubbleColorOutgoingUnsent
             case .sending:
-                return self.bubbleColorOutgoingSending
+                return bubbleColorOutgoingSending
             default:
-                return self.bubbleColorOutgoingSent
+                return bubbleColorOutgoingSent
             }
         } else {
             owsFail("Unexpected message type: \(message)")
-            return UIColor.ows_materialBlue
+            return bubbleColorOutgoingSent
         }
     }
 
     @objc
     public func bubbleColor(call: TSCall) -> UIColor {
-        if call.isIncoming {
-            return primaryColor
+        return bubbleColor(isIncoming: call.isIncoming)
+    }
+
+    @objc
+    public func bubbleColor(isIncoming: Bool) -> UIColor {
+        if isIncoming {
+            return ConversationStyle.defaultBubbleColorIncoming
         } else {
             return self.bubbleColorOutgoingSent
         }
     }
 
     @objc
-    public static var bubbleTextColorIncoming = UIColor.ows_white
+    public static var bubbleTextColorIncoming = UIColor.ows_black
+    public static var bubbleTextColorOutgoing = UIColor.ows_white
 
     @objc
     public func bubbleTextColor(message: TSMessage) -> UIColor {
         if message is TSIncomingMessage {
             return ConversationStyle.bubbleTextColorIncoming
-        } else if let outgoingMessage = message as? TSOutgoingMessage {
-            switch outgoingMessage.messageState {
-            case .failed:
-                return UIColor.ows_black
-            case .sending:
-                return UIColor.ows_black
-            default:
-                return UIColor.ows_black
-            }
+        } else if let _ = message as? TSOutgoingMessage {
+            return ConversationStyle.bubbleTextColorOutgoing
         } else {
             owsFail("Unexpected message type: \(message)")
-            return UIColor.ows_materialBlue
+            return ConversationStyle.bubbleTextColorOutgoing
         }
     }
 
     @objc
     public func bubbleTextColor(call: TSCall) -> UIColor {
-        if call.isIncoming {
+        return bubbleTextColor(isIncoming: call.isIncoming)
+    }
+
+    @objc
+    public func bubbleTextColor(isIncoming: Bool) -> UIColor {
+        if isIncoming {
             return ConversationStyle.bubbleTextColorIncoming
         } else {
-            return UIColor.ows_black
+            return ConversationStyle.bubbleTextColorOutgoing
         }
     }
+
+    @objc
+    public func bubbleSecondaryTextColor(isIncoming: Bool) -> UIColor {
+        return bubbleTextColor(isIncoming: isIncoming).withAlphaComponent(0.7)
+    }
+
+    @objc
+    public func quotedReplyBubbleColor(isIncoming: Bool) -> UIColor {
+        // TODO: Values
+        // TODO: Should this reflect "quoting self" v. "quoting other"?
```

Comment:
> // TODO: Should this reflect "quoting self" v. "quoting other"?

Is this still TODO? Seems like you addressed this with `quotingSelfHighlightColor`

Reply:
This has been changed to reflect conversation with Myles.